packages/cadl-ranch-specs/http/azure/core/page/main.tsp

```diff
@@ -0,0 +1,199 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/cadl-ranch-expect";
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+
+using Azure.Core;
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "For spec"
+@doc("Illustrates bodies templated with Azure Core with paging support")
+@scenarioService(
+  "/azure/core/page",
+  {
+    versioned: Versions,
+  }
+)
+namespace _Specs_.Azure.Core.Page;
+
+@doc("The version of the API.")
+enum Versions {
+  @doc("The version 2022-12-01-preview.")
+  @useDependency(global.Azure.Core.Versions.v1_0_Preview_2)
+  v2022_12_01_preview: "2022-12-01-preview",
+}
+
+@resource("users")
+@doc("Details about a user.")
+model User {
+  @key
+  @doc("The user's id.")
+  @visibility("read")
+  id: int32;
+
+  @doc("The user's name.")
+  name: string;
+
+  @doc("The user's order list")
+  orders?: UserOrder[];
+
+  ...global.Azure.Core.EtagProperty;
+}
+
+@doc("UserOrder for testing list with expand.")
+@resource("user")
+model UserOrder {
+  @key
+  @doc("The user's id.")
+  @visibility("read")
+  id: int32;
+
+  @doc("The user's id.")
+  userId: int32;
+
+  @doc("The user's order detail")
+  detail: string;
+}
+
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "For testing global.Azure.Core.Page"
+@scenario
+@doc("List with Azure.Core.Page<>.")
+@route("/page")
+@scenarioDoc("""
+Should only generate models named User and UserOrder.
+
+Should not generate visible model like Page.
+
+Expected query parameter: api-version=2022-12-01-preview
+
+Expected response body:
+```json
+{
+  "value":[
+     {
+        "id":1,
+        "name":"Madge",
+        "etag": "11bdc430-65e8-45ad-81d9-8ffa60d55b59"
+     }
+  ]
+}
+""")
+op listWithPage is global.Azure.Core.Foundations.Operation<{}, global.Azure.Core.Page<User>>;
+
+@doc("The parameters for listing users.")
+model ListItemInput {
+  @doc("The body of the input.")
+  @body
+  bodyInput: ListItemInputBody;
```

Comment:
Missed in previous review. I don't think we should put this aviod-rfc thing in the cadl-ranch test.

Reply:
No you didn't. We discussed that. It is in ledger entry.